### PR TITLE
chore: migrate from tslint to eslint

### LINKED ISSUE #248: Migrate from tslint to eslint
### Is your feature request related to a problem? Please describe.
eslint is recommended instead of tslint

### Describe the solution you'd like
Migrate and revise rules

### PR DESCRIPTION
**Pull request by P1NHE4D**
⚙️ **Changes:**
* removed tslint packages and files
* added eslint

🎟️ **Closes #248**

### CODE REVIEW

**Comment by janivo** [comment]
Hey @P1NHE4D, is this WIP, or is it ready for review?

**Comment by P1NHE4D** [comment]
> Hey @P1NHE4D, is this WIP, or is it ready for review?

It's WIP for now due to the failing pipeline. I'll let you know once I fixed it.

File: packages/elements/src/components/ino-datepicker/local.ts [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 // Supported Languages
-export type DatepickerLocale = 'de'|'en';
+export type DatepickerLocale = "de"|"en";
 
 // Map: Language Key => Flatpickr language
 export const datepickerLocales: Record<DatepickerLocale, CustomLocale> = {
-  'de': German,
-  'en': English
-}
+	"de": German,
```
Comment (janivo) [inline packages/elements/src/components/ino-datepicker/local.ts:10]: This seems to collide with our prettier rules which formats every `""` to `''`

**Review by janivo — CHANGES_REQUESTED** [review]
Hey @AlessaRad, could you check this PR as well?
I think you might be more experienced on this topic.

File: packages/elements/src/components/ino-datepicker/local.ts (hunk 1 above)
Comment (P1NHE4D) [reply packages/elements/src/components/ino-datepicker/local.ts:10]: Oh okay I didn't realize that we're using single quotes for some reason - I'll look into it

Comment (P1NHE4D) [reply packages/elements/src/components/ino-datepicker/local.ts]: fixed ✅

**Comment by AlessaRad** [comment]
Sure, I can take a look at it!

File: packages/elements/src/components/ino-icon/validate.ts [hunk 2]
```diff
@@ -1,54 +1,54 @@
...
+		}
 
-    // tslint:disable-next-line:prefer-for-of
-    for (let i = 0; i < elm.attributes.length; i++) {
-      const val = elm.attributes[i].value;
-      if (isStr(val) && val.toLowerCase().indexOf('on') === 0) {
-        return false;
-      }
-    }
+		// tslint:disable-next-line:prefer-for-of
```
Comment (AlessaRad) [inline packages/elements/src/components/ino-icon/validate.ts:38]: Search for 'tslint' and 'tslint.json' files in the entire project and check if occurrences still needed

File: package.json [hunk 3]
```diff
@@ -15,6 +15,7 @@
     "clean:root": "shx rm -rf node_modules",
     "update:commit-scopes": "node scripts/generateCommitScopeEnum.js",
     "lint": "lerna run lint",
```
Comment (AlessaRad) [inline package.json:17]: yarn lint still exits with errors. For a better overview you can execute in in the respective packages

File: package.json [hunk 4]
```diff
@@ -61,4 +62,4 @@
   },
   "name": "core",
   "version": "2.3.0"
-}
\ No newline at end of file
+}
```
Comment (AlessaRad) [inline package.json:65]: Make sure, that the pre-commit hook uses eslint to check the changed codebase (see `.lintstagedrc`file)

**Review by AlessaRad — COMMENTED** [review]
Hey, I found a couple of things that we should have as second look at!

File: packages/elements/src/components/ino-icon/validate.ts (hunk 2 above)
Comment (P1NHE4D) [reply packages/elements/src/components/ino-icon/validate.ts]: Thank you for your extensive review! I just implemented your feedback - let me know if you have any other suggestions.

**Comment by janivo** [comment]
Hey @P1NHE4D, could you resolve this conflict?

**Comment by P1NHE4D** [comment]
> Hey @P1NHE4D, could you resolve this conflict?

Done ✅

**Comment by AlessaRad** [comment]
Hi @P1NHE4D , I reviewed your latest changes :) No more linting errors, great! The only thing I noticed is that the pre-commit hook is still not working properly. If there is a linting error, the git log only says "Parsing error: Cannot read file '/.../tsconfig.json'". Ideally it should print the exact code line where an eslint rule was violated.

**Comment by janivo** [comment]
Hey @P1NHE4D, could you resolve these conflicts?

**Review by janivo — APPROVED** [review]

**Pull request merged**